src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -88,15 +104,276 @@ export class PtyService extends Disposable implements IPtyService {
 		return this._throwIfNoPty(id).getCwd();
 	}
 
-	async getLatency(id: number): Promise<number> {
-		return this._throwIfNoPty(id).getLatency();
+	async setTerminalLayoutInfo(args: ISetTerminalLayoutInfoArgs): Promise<void> {
+		this._workspaceLayoutInfos.set(args.workspaceId, args);
+	}
+
+	async getTerminalLayoutInfo(): Promise<ITerminalsLayoutInfo | undefined> {
+		const layout = this._workspaceLayoutInfos.get(this._workspaceContextService.getWorkspace().id);
```

Comment:
Let's pass the workspaceId over instead of trying to use the workspace context service here since it doesn't exist.